From 7a69c41f0b93d14bc7ca8bf9e4d49091fd36bace Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Thu, 12 Jul 2012 11:01:47 +0300
Subject: [PATCH 235/311] confc.h: look for specific service uuid in recipe3

An example with searching for devices used by a particular service
instance is more realistic than searching for devices used by a
particular service type.

LogD #395
RB: r/865
---
 conf/confc.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/conf/confc.h b/conf/confc.h
index 2917a46..4ed19c2 100644
--- a/conf/confc.h
+++ b/conf/confc.h
@@ -245,8 +245,7 @@ struct c2_mutex;
  *
  * /// Accesses configuration data of devices used by specific service on
  * /// specific node.
- * static int specific_devices_process(enum c2_cfg_service_type svc_type,
- *                                     const char *node_id)
+ * static int specific_devices_process(const char *svc_id, const char *node_id)
  * {
  *         struct c2_conf_obj   *dir_obj;
  *         struct c2_conf_dircur dir;
@@ -263,11 +262,9 @@ struct c2_mutex;
  *         // take care of closing svc_obj.
  *
  *         while ((rc = c2_confc_readdir_sync(&svc_obj, &dir)) > 0) {
- *                 struct c2_conf_obj     *node_obj;
- *                 struct c2_conf_service *svc =
- *                         C2_CONF_CAST(svc_obj, c2_conf_service);
+ *                 struct c2_conf_obj *node_obj;
  *
- *                 if (svc->cs_type != svc_type)
+ *                 if (!streq(svc_obj->co_id, svc_id))
  *                         // This is not the service we are looking for.
  *                         continue;
  *
-- 
1.8.3.2

